### PR TITLE
Reset service-mirror component when target's k8s API is unreachable

### PR DESCRIPTION
**Pull request by alpeb**
When the service-mirror component can't reach the target's k8s API, the goroutine blocks and it can't be unblocked.

This was happenining specifically  in the case of the multicluster integration test (still to be pushed), where the source and target clusters are created in quick succession and the target's API service doesn't always have time to be exposed before being requested by the service mirror.

The fix consists on no longer have `restartClusterWatcher` be side-effecting, and instead return an error. If such error is not nil then the link watcher is stopped and reset after 10 seconds.

### CODE REVIEW

File: controller/cmd/service-mirror/main.go [hunk 1]
```diff
@@ -107,7 +107,14 @@ main:
 							if err != nil {
 								log.Errorf("Failed to load remote cluster credentials: %s", err)
 							}
-							restartClusterWatcher(link, *namespace, creds, controllerK8sAPI, *requeueLimit, *repairPeriod, metrics)
+							err = restartClusterWatcher(link, *namespace, creds, controllerK8sAPI, *requeueLimit, *repairPeriod, metrics)
+							if err != nil {
+								// failed to restart cluster watcher; give a bit of slack
+								// and restart the link watch to give it another try
+								log.Error(err)
+								time.Sleep(10 * time.Second)
```
Comment (zaharidichev) [inline controller/cmd/service-mirror/main.go:115]: Nit: may want to put that in a `const`

**Review by zaharidichev — APPROVED** [review]
LGTM

**Comment by kleimkuhler** [comment]
The base branch for this should probably be `main` since it's fixing an issue, even if it is related to adding the multicluster integration test.

**Review by kleimkuhler — APPROVED** [review]
Looks good 👍

**Comment by alpeb** [comment]
> The base branch for this should probably be `main` since it's fixing an issue, even if it is related to adding the multicluster integration test.

Yeah, the reason for doing the daisy chain PRs was for all to pass the tests. Now that this one has been reviewed and doesn't have dependencies on the changes from the parent branches, I'll merge to main and rebase the other ones :+1:

File: controller/cmd/service-mirror/main.go (hunk 1 above)
Comment (alpeb) [reply controller/cmd/service-mirror/main.go:115]: will do :+1:

**Comment by kleimkuhler** [comment]
@alpeb Sounds good!

**Pull request merged**